resources/js/vat-validation.js

```diff
@@ -0,0 +1,123 @@
+import { useThrottleFn } from '@vueuse/core'
+import { token } from 'Vendor/rapidez/core/resources/js/stores/useUser'
+import { mask } from 'Vendor/rapidez/core/resources/js/stores/useMask'
+
+const patterns = {
+    AT: /^U[A-Z\d]{8}$/,
+    BE: /^[0-2]\d{9}$/,
+    BG: /^\d{9,10}$/,
+    CY: /^\d{8}[A-Z]$/,
+    CZ: /^\d{8,10}$/,
+    DE: /^\d{9}$/,
+    DK: /^(\d{2} ?){3}\d{2}$/,
+    EE: /^\d{9}$/,
+    EL: /^\d{9}$/,
+    ES: /^([A-Z]\d{7}[A-Z]|\d{8}[A-Z]|[A-Z]\d{8})$/,
+    EU: /^\d{9}$/,
+    FI: /^\d{8}$/,
+    FR: /^[A-Z\d]{2}\d{9}$/,
+    GB: /^(\d{9}|\d{12}|(GD|HA)\d{3})$/,
+    HR: /^\d{11}$/,
+    HU: /^\d{8}$/,
+    IE: /^([A-Z\d]{8}|[A-Z\d]{9})$/,
+    IT: /^\d{11}$/,
+    LT: /^(\d{9}|\d{12})$/,
+    LU: /^\d{8}$/,
+    LV: /^\d{11}$/,
+    MT: /^\d{8}$/,
+    NL: /^\d{9}B\d{2}$/,
+    PL: /^\d{10}$/,
+    PT: /^\d{9}$/,
+    RO: /^\d{2,10}$/,
+    SE: /^\d{12}$/,
+    SI: /^\d{8}$/,
+    SK: /^\d{10}$/,
+    SM: /^\d{5}$/,
+}
+
+document.addEventListener('turbo:load', function () {
+    window.app.$on('vat-change', async (event) => {
+        let cleanVatid = event.target.value.replace(/[\s\.-]/g, '')
+
+        if(event.target.value != cleanVatid) {
```

Comment:
Am i correct in seeing all browser validation is ignored during this check and we only do our checks? (until the form is actually submitted)

Maybe we should check the input's validity first before doing our more expensive checks, so more basic checks like the `required`, `minlength`, `maxlength`, `type`, and `pattern` checks which come built in are done first.

Reply:
I've added a check, but I will say that I wouldn't want any of those checks in the core at the very least, as for example a valid number can range from 7 to 15 characters long and have a lot of different patterns :)